gui/layers/_marker.py

```diff
@@ -0,0 +1,124 @@
+import weakref
+
+import numpy as np
+
+from ._base import Layer
+from .._vispy.scene.visuals import Markers as MarkersNode
+
+from ._register import add_to_viewer
+
+
+@add_to_viewer
+class Marker(Layer):
+    """Marker layer.
+
+    Parameters
+    ----------
+    marker_coords : np.ndarray
+        coordinates for each marker.
+
+    marker_args: Dict
+        keyword arguments for the marker style.
+        see set_data():
+        http://api.vispy.org/en/latest/visuals.html#vispy.visuals.MarkersVisual
+
+    """
+
+    def __init__(self, marker_coords, marker_args={}):
+
+        visual = MarkersNode()
+        super().__init__(visual)
+
+        # Save the marker coordinates
+        self._marker_coords = marker_coords
+
+        self._marker_args = marker_args
+
+        # update flags
+        self._need_display_update = False
+        self._need_visual_update = False
+
+    @property
+    def marker_coords(self):
+        """ndarray: coordinates of the marker centroids
+        """
+        return self._marker_coords
+
+    @property
+    def data(self):
+        """ndarray: coordinates of the marker centroids
+        """
+        return self._marker_coords
+
+    @data.setter
+    def data(self, data):
+        self._marker_coords = data
+        self.refresh()
+
+    @property
+    def marker_args(self):
```

Comment:
Would prefer to have properties for each individual argument instead of this, and to keep `marker_args` "private."

Reply:
@sofroniewn pointed this out as well and I definitely agree. I implemented this.